aws/ecs/packer/to-upload/peers.sh

```diff
@@ -1,30 +1,225 @@
 #!/bin/bash
-# This script print a list of IPs of the EC2 instances in your current Autoscaling Group.
+# This script print a list of IPs based on
+#   1) TAG: currently defaulted to WEAVE_ECS, this will match IPs for instances that have matching values for that Tag key
+#   2) Autoscaling Group: if this instance is part of an autoscaling group, this will return IPs for its peers
+#   3) Spot Fleet: Finally, if this instance is part of a spot fleet, this will return IPs for all instances in the fleet
 set -eu
 
-aws=/usr/local/bin/aws
+__instance_tag_key="${1:-WEAVE_ECS}"
 
-export AWS_DEFAULT_REGION=$(curl -s curl http://169.254.169.254/latest/dynamic/instance-identity/document | \
-                            jq -r .region)
-current_instance_id=$(curl -s curl http://169.254.169.254/latest/meta-data/instance-id)
-current_autoscaling_group=$($aws autoscaling describe-auto-scaling-instances | \
-                            jq -r ".AutoScalingInstances[] | select(.InstanceId == \"$current_instance_id\") | .AutoScalingGroupName")
+__awscli=
+__instance_id=
+__region=
+__instance_autoscaling_group=
+__peer_autoscaling_instance_ids=
+__list_of_active_spot_fleets=
+__spot_fleet_request_id=
+__peer_spot_fleet_instance_ids=
+__peer_private_ips=
+__peer_tag_instance_ids=
 
-if [ -z "$current_autoscaling_group" ]; then
-  >&2 echo "instance doesn't belong to an autoscaling group"
-  exit 1
-fi
+check_data() {
+  if [[ "${1}x" == "x" ]]; then
+    return 1
+  fi
+}
 
-peer_instances=$($aws autoscaling describe-auto-scaling-instances | \
-                jq -r ".AutoScalingInstances[] | select(.AutoScalingGroupName == \"$current_autoscaling_group\" and .InstanceId != \"$current_instance_id\") | .InstanceId")
+check_exit() {
+  if [[ "${1}" != "0" ]]; then
+    >&2 echo "${FUNCNAME[1]} Exited with a nonzero status"
+    exit 2
+  fi
+}
 
-if [ -z "$peer_instances" ]; then
-# no peers found
-  exit 0
-fi
+get_from_curl() {
+  local __endpoint="${1}"
+  local __retry_attempts=3
+  local __retry_delay=1
+  local __retries=0
+  local __retval=
+  # the metadata service occasionally returns empty with a 200 status, so retry a couple times just in case (very rare)
+  while true; do
+    local __retval=$(curl -s "http://169.254.169.254/latest/${__endpoint}")
+    if [[ "${__retval}x" != "x" ]]; then
+      break
+    else
+      sleep "${__retry_delay}"
+      __retries=$(( __retries + 1 ))
+      if [[ "${__retries}" -eq "${__retry_attempts}" ]]; then
+        >&2 echo "${FUNCNAME[0]} WARNING: Curl could not reach metadata for latest/${__endpoint}"
+        exit # Not necessarily a problem but bail out, i.e. reboot: https://github.com/weaveworks/integrations/issues/20
+      fi
+    fi
+  done
+  printf "${__retval}"
+}
+
+get_region() {
+  if ! check_data "${__region}"; then
+    __region="$(get_from_curl "dynamic/instance-identity/document" | jq -r .region; check_exit "${PIPESTATUS[1]}")"
```

Comment:
pulling from a different metadata endpoint (the availability zone) i got rid of the jq dependency completely